src/NuGet.Core/NuGet.Protocol/Resources/PackageUpdateResource.cs

```diff
@@ -430,11 +430,26 @@ private static string GetSymbolsPath(string packagePath, bool isSnupkg)
             }
             else
             {
-                 AdvertiseAvailableOptionToIgnore(response.StatusCode, logger);
+                AdvertiseAvailableOptionToIgnore(response.StatusCode, logger);
             }
 
             //No exception to the rule specified.
-            response.EnsureSuccessStatusCode();
+
+#if !NETCOREAPP5_0
+            // Before calling EnsureSuccessStatusCode(), squirrel away statuscode, in order to add it to exception.
+            HttpStatusCode statusCode = response.StatusCode;
+            try
+            {
+#endif
+                response.EnsureSuccessStatusCode();
+#if !NETCOREAPP5_0
+            }
+            catch (HttpRequestException ex)
+            {
+                ex.Data["StatusCode"] = statusCode;
+                throw;
+            }
```

Comment:
Looks like code duplicated from `HttpSource`.  Can you extract the code into an internal helper method (like in `HttpRequestExceptionUtility`) and call it in both places?

Reply:
yep. next iteration.